detect_secrets/plugins/ibm_cloud_iam.py

```diff
@@ -34,6 +35,7 @@ def verify(self, secret: str) -> VerifiedResult:
             else VerifiedResult.VERIFIED_FALSE
 
 
+@lru_cache()
```

Comment:
I don't think we should enforce caching without given caller an option. Here are my reasons 

1. IBM Cloud token supports [lock/unlock an API key](https://cloud.ibm.com/docs/account?topic=account-userapikey#lock_user_key). You can get different results for the same key
2. detect-secrets plugins can be used as a library. We internally using it as a supporting library for a long run application to continuously scanning tokens. Mandating the cache will prevent us validate a token when it has switched from locked to unlocked state.

I would suggest we either filter off some keywords or use filter plugins during scan to avoid false positives, or provide a way to allow caller to use verify without using cache (solve case 2 ☝️ ).

Reply:
killuazhu++